requirements.txt

```diff
@@ -5,6 +5,7 @@ torch>=1.3,<1.8
 future>=0.17.1  # required for builtins in setup.py
 # pyyaml>=3.13
 PyYAML>=5.1  # OmegaConf requirement >=5.1
+jsonargparse[signatures]>=3.0.0.dev4  # New trainer_cli requirement
```

Comment:
using `dev4` does not look very stable...

Reply:
Version 3.0.0 will be released very soon. It is still in dev mostly waiting for the feedback from this pull request. It is also already highly tested, see https://codecov.io/gh/omni-us/jsonargparse/src/master/jsonargparse/signatures.py

Reply:
can we use also version 2?

Reply:
No, version 2 does not include the main feature used for this. Still, if people like this PR we don't need to rush it. We could first try it out with different models and use cases and extending jsonargparse where needed. Only after it is stable and highly tested we include this in lightning.

Reply:
The version is now `3.0.0rc1`. Still a bit more of work remaining for the official `3.0.0` release.

Reply:
Version 3 has now been released.